tests/h/views/api_groups_test.py

```diff
@@ -8,28 +8,36 @@
 from pyramid.httpexceptions import HTTPNoContent, HTTPBadRequest
 
 from h.views import api_groups as views
+from h.services.profile_group import ProfileGroupService
+from h.services.group import GroupService
 
 
 @pytest.mark.usefixtures('profile_group_service')
 class TestGroups(object):
     def test_groups_proxies_to_service(self, pyramid_request, profile_group_service):
         views.groups(pyramid_request)
 
-        assert profile_group_service.called_once()
+        profile_group_service.all.assert_called_once()
 
     def test_groups_passes_authority_parameter(self, pyramid_request, profile_group_service):
         pyramid_request.params = {'authority': 'foo.com'}
 
         views.groups(pyramid_request)
 
-        assert profile_group_service.called_once_with(pyramid_request, 'foo.com')
+        profile_group_service.all.assert_called_once_with(user=mock.ANY,
+                                                          authority='foo.com',
+                                                          document_uri=mock.ANY)
+
+        c_args, c_kwargs = profile_group_service.all.call_args
+        assert c_kwargs['authority'] == 'foo.com'
 
     def test_groups_passes_document_uri_parameter(self, pyramid_request, profile_group_service):
         pyramid_request.params = {'document_uri': 'foo.example.com'}
 
         views.groups(pyramid_request)
 
-        assert profile_group_service.called_once_with(pyramid_request, document_uri='foo.example.com')
+        c_args, c_kwargs = profile_group_service.all.call_args
```

Comment:
What is the rationale for checking `call_args` here rather than `assert_called_once_with` and `mock.Any` for other parameters as above?

Reply:
This test is concerned with whether `document_uri` got passed to the service method. The method in question takes, I think, 4 possible arguments. As the test's concern is just that one argument, I thought it might be more "focused" to test for it alone. `called_once_with` requires one to itemize _all_ args that get passed.